packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -148,7 +150,9 @@ export function createCloudflareEnvironmentOptions(
 					//       so the input value here serves both as the build input as well as the starting point for
 					//       dev pre-bundling crawling (were we not to set this input field we'd have to appropriately set
 					//       optimizeDeps.entries in the dev config)
-					input: options.main,
+					input: userConfig.root
+						? path.resolve(userConfig.root, options.main)
+						: path.resolve(options.main),
```

Comment:
It turns out that `rollupOptions.input` isn't automatically resolved relative to the root so I had to add the resolution here.